Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift

```diff
@@ -122,19 +123,20 @@ import PayPalCheckout
                         case .vault:
                             action.set(billingAgreementToken: order.orderID)
                         @unknown default:
+                            self?.apiClient.sendAnalyticsEvent(BTPayPalNativeCheckoutAnalytics.createOrderInvalidPaymentType)
+                            self?.apiClient.sendAnalyticsEvent(BTPayPalNativeCheckoutAnalytics.tokenizeFailed)
                             completion(nil, BTPayPalNativeError.invalidRequest)
                         }
                     },
                     onApprove: { [weak self] approval in
-                        self?.apiClient.sendAnalyticsEvent("ios.paypal-native.on-approve.started")
                         self?.tokenize(approval: approval, request: request, completion: completion)
                     },
                     onCancel: {
-                        self?.apiClient.sendAnalyticsEvent("ios.paypal-native.canceled")
+                        self?.apiClient.sendAnalyticsEvent(BTPayPalNativeCheckoutAnalytics.tokenizeCanceled)
                         completion(nil, BTPayPalNativeError.canceled)
                     },
                     onError: { error in
-                        self?.apiClient.sendAnalyticsEvent("ios.paypal-native.on-error.failed")
```

Comment:
Do we need separate "on-error" messages?

Reply:
🤔 we may want to return something different than the tokenize failure here, but maybe something like `ios.paypal-native.failed` to match the above cancel case

Reply:
Technically this is pre tokenize, so feels odd to send `tokenizeFailed` imo - curious to hear @scannillo thoughts though.

Reply:
The flow is a bit different with Native checkout for me, in that there is the umbrella `tokenize` where `createOrder` is called, and upon success, you call the another `tokenize` function that takes a request AND an approval and performs the completion with the nonce passed in from umbrella `tokenize` function

Reply:
So I if there is a `tokenize:start`, there should be `tokenize:failed` or` tokenize:success` end event. That is my understanding so far.

Reply:
> Technically this is pre tokenize, so feels odd to send tokenizeFailed imo - curious to hear @scannillo thoughts though.

All routes should end in 1 of 3 cases - success, fail, cancel. If we fail to get an orderID, or of NXO returns us an error, our `tokenize` function `fails` to the merchant.

Reply:
> So I if there is a tokenize:start, there should be tokenize:failed or tokenize:success or tokenize:canceled (in some cases) end event. That is my understanding so far.

I agree with this!

Reply:
For each start event, we should also sent 1 out of the 3 possible end events - fail, success, cancel.

Reply:
That makes sense 👍